### PR TITLE
Unhardcode pycodestyle ignored tests

### LINKED ISSUE #700: Allow pycodestyle checks to be disabled
Currently the `PycodestyleChecker` hard-codes a list of pycodestyle errors to disable. We should make this a PythonTA configuration option, under a new section `[PYCODESTYLE]` (see `.pylintrc`).

### PR DESCRIPTION
**Pull request by simonchenwastaken**
## Motivation and Context
Removes the hard-coding in `python_ta/checkers/pycodestyle_checker.py` into a config folder. Allows for these to be disabled or changed easily in configuration.

(resolves #700)
## Your Changes

**Description**:

Change pycodestyle ignored tests from a literal in the custom checker `pycodestyle_checker.py` into a pylint config option in `python_ta/.pylintrc`. Add config to `pycodestyle_checker.py` . Config option is written as a csv.

**Type of change**:

- [x] Refactoring (internal change to codebase, without changing functionality)

## Testing

- Manual testing. A temporary code snippet like the one was used to check if the configuration was properly set during experimentation:
```python
print(linter.config.pycodestyle_ignored_tests)
```
- Ran pytest

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have verified that the TravisCI tests have passed. 
- [x] I have reviewed the test coverage changes reported on Coveralls.

### CODE REVIEW

File: python_ta/.pylintrc [hunk 1]
```diff
@@ -101,3 +101,24 @@ argument-rgx = (([a-z][a-z0-9_]{0,30})|(_[a-z0-9_]*))$
 class-rgx    = (([A-Z][a-zA-Z0-9]{0,30})|(_[A-Z][a-zA-Z0-9]*))$
 method-rgx   = (([a-z][a-z0-9_]{0,30})|(_[a-z0-9_]*))$
 class-attribute-rgx = ([A-Za-z_][A-Za-z0-9_]{0,30}|(__.*__))$
+
+[PYCODESTYLE]
+# Ignored pycodestyle checks
+pycodestyle-ignored-tests =
```
Comment (david-yz-liu) [inline python_ta/.pylintrc:107]: Rename this option to `pycodestyle-ignore` (it's more consistent with how pycodestyle names it).

File: python_ta/.pylintrc [hunk 2]
```diff
@@ -101,3 +101,24 @@ argument-rgx = (([a-z][a-z0-9_]{0,30})|(_[a-z0-9_]*))$
...
+    E702,
+    E703,
+    E711,
+    E712,
+    E722,
+    W291,
+    W292,
+    W293,
+    W391,
+    W503
```
Comment (david-yz-liu) [inline python_ta/.pylintrc:124]: Make sure to end the file with a new line.

File: python_ta/__init__.py [hunk 3]
```diff
@@ -212,7 +212,12 @@ def reset_linter(config=None, file_linted=None):
          {'default': 'http://127.0.0.1:5000',
           'type': 'string',
           'metavar': '<server-url>',
-          'help': 'Server address to submit anonymous data'})
+          'help': 'Server address to submit anonymous data'}),
+        ('pycodestyle-ignored-tests',
```
Comment (david-yz-liu) [inline python_ta/__init__.py:216]: This is the right option to add, but you should add it directly to the checker, not here. Some of the other custom checkers do this, if you want to take a look.

File: python_ta/.pylintrc [hunk 4]
```diff
@@ -101,3 +101,24 @@ argument-rgx = (([a-z][a-z0-9_]{0,30})|(_[a-z0-9_]*))$
 class-rgx    = (([A-Z][a-zA-Z0-9]{0,30})|(_[A-Z][a-zA-Z0-9]*))$
 method-rgx   = (([a-z][a-z0-9_]{0,30})|(_[a-z0-9_]*))$
 class-attribute-rgx = ([A-Za-z_][A-Za-z0-9_]{0,30}|(__.*__))$
+
+[PYCODESTYLE]
+# Ignored pycodestyle checks
+pycodestyle-ignored-tests =
+    E111,
```
Comment (david-yz-liu) [inline python_ta/.pylintrc:108]: You should keep the comments from the original list.

**Review by david-yz-liu — COMMENTED** [review]
@simonchenwastaken good work. I just spent some time testing this out and it looks like after you moved the option into the checker, it's not being processed properly by pylint. After some digging, I discovered that the reason is *the checker needs to be loaded before the config options are processed in `python_ta/__init__.py`.

So you should make the following changes in `__init__.py`:

1. (`__init__.py`) Move this checker into the list of other custom checkers being registered.
2. (`__init__.py`) Remove the `pyta_pep8` configuration option, so that the checker is always loaded.

**Comment by simonchenwastaken** [comment]
I believe these are the 2 changes you asked for 👍

**Comment by david-yz-liu** [comment]
@simonchenwastaken good, but you should also remove the option entirely since it's no longer being used.

File: python_ta/.pylintrc [hunk 5]
```diff
@@ -5,9 +5,7 @@
 # Default max amount of messages for reporter to display.
 pyta-number-of-messages = 5
 
-# Set whether to use the pycodestyle and type-checking checkers.
-# Anything for a True/False, <yn> value should use in config file: y, yes, n, no
-pyta-pep8 = yes
+# Set whether to use the pycodestyle type-checking checkers.
```
Comment (david-yz-liu) [inline python_ta/.pylintrc:8]: Remove the word "pycodestyle" from this sentence, and also the word "checkers" should now be singular.

**Review by david-yz-liu — COMMENTED** [review]
@simonchenwastaken did you try running just `python_ta.check_all` on a separate file? Looks like there's an error. (Always make sure to do this whenever you make a commit.)

**Comment by simonchenwastaken** [comment]
> @simonchenwastaken did you try running just `python_ta.check_all` on a separate file? Looks like there's an error. (Always make sure to do this whenever you make a commit.)

I will be more careful with my commits on undrafted PRs in the future. I made sure to run both `pytest` and `python_ta.check_all()' before this last commit and I think everything is okay

**Pull request merged**